V2/src/fc_state_machine.cpp

```diff
@@ -177,6 +178,21 @@ void fc_state_machine_thread()
                 update_leds();
             }
         }
+        else if (FC_PURGE == state)
+        {
+            if (purgeTimer.read() > 0.2)
+            {
+                purge_v.write(false);
+                purgeTimer.stop();
+                purgeTimer.reset();
+            }
+            if ((!purge_v)&(get_analog_values().press1>PRESSURE_MIN))
```

Comment:
We should add a timeout to this check in case we run out of hydrogen in purge state